packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -642,24 +663,35 @@ class _MonthPickerState extends State<MonthPicker> {
         children: <Widget>[
           new Semantics(
             sortKey: _MonthPickerSortKey.calendar,
-            child: new PageView.builder(
-              key: new ValueKey<DateTime>(widget.selectedDate),
-              controller: _dayPickerController,
-              scrollDirection: Axis.horizontal,
-              itemCount: _monthDelta(widget.firstDate, widget.lastDate) + 1,
-              itemBuilder: _buildItems,
-              onPageChanged: _handleMonthPageChanged,
+            child: new NotificationListener<ScrollStartNotification>(
+              onNotification: (_) => _handleScrollStartNotification(context),
+              child: new NotificationListener<ScrollEndNotification>(
+                onNotification: (_) => _handleScrollEndNotification(context),
+                child: new PageView.builder(
+                  key: new ValueKey<DateTime>(widget.selectedDate),
+                  controller: _dayPickerController,
+                  scrollDirection: Axis.horizontal,
+                  itemCount: _monthDelta(widget.firstDate, widget.lastDate) + 1,
+                  itemBuilder: _buildItems,
+                  onPageChanged: _handleMonthPageChanged,
+                ),
+              ),
             ),
           ),
           new PositionedDirectional(
             top: 0.0,
             start: 8.0,
             child: new Semantics(
               sortKey: _MonthPickerSortKey.previousMonth,
-              child: new IconButton(
-                icon: const Icon(Icons.chevron_left),
-                tooltip: _isDisplayingFirstMonth ? null : '${localizations.previousMonthTooltip} ${localizations.formatMonthYear(_previousMonthDate)}',
-                onPressed: _isDisplayingFirstMonth ? null : _handlePreviousMonth,
+              child: new FadeTransition(
+                opacity: _chevronOpacityAnimation,
+                child: new IconButton(
+                  icon: new Icon(Icons.chevron_left),
+                  tooltip: _isDisplayingFirstMonth
```

Comment:
More nit pickery. The way this was formatted helped highlight the fact that tooltip and onPressed values had similar structure. If you're going to do this, then 2-space indent.

Reply:
Auto-formatter strikes; will revert to previous formatting